clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1276,7 +1276,12 @@ void CodeGenModule::EmitCtorList(CtorList &Fns, const char *GlobalName) {
     ctor.addInt(Int32Ty, I.Priority);
     ctor.add(llvm::ConstantExpr::getBitCast(I.Initializer, CtorPFTy));
     if (I.AssociatedData)
-      ctor.add(llvm::ConstantExpr::getBitCast(I.AssociatedData, VoidPtrTy));
+      if (I.AssociatedData->getType()->getPointerAddressSpace() !=
+          VoidPtrTy->getAddressSpace())
```

Comment:
Should we check that it is correct to do address space cast here?
Like this check
https://github.com/intel/llvm/blob/ccefc93c884556ca1de01cfe2184274ea7c1d2f7/clang/lib/CodeGen/CGExprScalar.cpp#L2996
@asavonic ?

Reply:
Yes, we have to ensure that an addrspace cast is valid.

Reply:
Okay, will add this check.

Reply:
In the cases above, it looks like they values are both cast to the generic address space, to ensure sameness.   This does not seem to be needed here, as the cast is to address space 0.  However, I have changed the call to use an existing routine (which does the check as well) which is called in few other places, instead of checking it directly how I had it before.